src/renderers/fields/field.ts

```diff
@@ -0,0 +1,54 @@
+import { Component } from '../../common/binding';
+import { update } from '../../actions';
+import { JsonSchema } from '../../models/jsonSchema';
+import { UISchemaElement } from '../../models/uischema';
+export interface FieldProps {
+  /**
+   * The UI schema to be rendered.
+   */
+  uischema: UISchemaElement;
+
+  /**
+   * The JSON schema that describes the data.
+   */
+  schema: JsonSchema;
+  /**
+   * Optional instance path. Necessary when the actual data
+   * path can not be inferred via the UI schema element as
+   * it is the case with nested controls.
+   */
+  path?: string;
+  data: any;
+  className: string;
+  id: string;
+  visible: boolean;
+  enabled: boolean;
+  dispatch: any;
+}
+export interface FieldState {
+  value: any;
+}
+export class Field<P extends FieldProps, S extends FieldState> extends Component<P, S> {
+
+  constructor(props: P) {
+    super(props);
+    this.state = {
```

Comment:
I think we can get rid of `state` again as I don't see where we are using it anyway, since we read the `data` property on each render now instead from `this.state.value`. We hence should be able to remove `FieldState` entirely or keep it empty.